activemq-client/src/main/java/org/apache/activemq/util/JMSExceptionSupport.java

```diff
@@ -61,6 +63,11 @@ public static JMSException create(Exception cause) {
         if (cause instanceof JMSException) {
             return (JMSException)cause;
         }
+        if (cause instanceof MaxFrameSizeExceededException) {
+            JMSException jmsException = new JMSException(cause.getMessage(), "41300");
+            jmsException.setLinkedException(cause);
```

Comment:
Instead of using linked I would use initCause() as thats what the server side does.  A server side error thrown will have a TransportDisposedIOExceptionset when calling e.getCause() so it would be consistent

Reply:
Should we do both? A number of the JMSExceptionSupport hanlding is linking both in.  only costs a pointer

Reply:
My commit does both now